File: JULIA/Cap08/c08ex03.jl
# c08ex03.jl

println("CLASSIFICACAO DE NUMEROS (CRESCENTE)\n")

numeros = zeros(Int, 5)

for i in 1:5
  print("Entre o $(i)o. numero: ")
  numeros[i] = parse(Int, readline())
end

for i in 1:4
  for j in (i + 1):5
    if (numeros[i] > numeros[j])
      x = numeros[i]
      numeros[i] = numeros[j]
      numeros[j] = x
    end
  end
end

println()
for i in 1:5
  println("$(i)o. numero: $(numeros[i])")
end

print("\nPressione <Enter> para encerrar... ")
readline()

File: RUST/Cap08/c08ex03.rs
// c08ex03.rs

use std::io::{self, Write};

fn main() {
  println!("CLASSIFICACAO DE NUMEROS (CRESCENTE)\n");

  let mut numeros: Vec<i32> = Vec::with_capacity(5);

  for i in 0..=4 {
    print!("Entre o {}o. numero: ", i + 1);
    io::stdout().flush().expect("Falha ao limpar o buffer");

    let mut entra_n = String::new();
    io::stdin().read_line(&mut entra_n).expect("Falha ao ler a entrada");
    let numero: i32 = entra_n.trim().parse().expect("Entrada inválida");
    numeros.push(numero);
  }

  for i in 0..=3 {
    for j in (i + 1)..=4 {
      if numeros[i] > numeros[j] {
        let x = numeros[i];
        numeros[i] = numeros[j];
        numeros[j] = x;
      }
    }
  }

  println!();
  for i in 0..=4 {
    println!("{}o. numero: {}", i + 1, numeros[i]);
  }

  println!();
  print!("Pressione <Enter> para encerrar... ");
  io::stdout().flush().expect("Falha ao limpar o buffer");
  let mut _entrada = String::new();
  io::stdin().read_line(&mut _entrada).expect("Falha ao ler a entrada");
}
